Add sepolicy for npu libraries

akodanka opened the pull request.

JeevakaPrabu committed:
Add sepolicy rules to access npu from shell and app

Access to npu device fails from both android shell and application.

Changes done:
-Define npu_device as a mlstrustedobject
-Allow appdomain and shell to access npu_device
-Add npu dependent libraries to same_process_hal_file security context

Tests done:
-Run hello_query_device from adb shell
-Use npu device for inference using openvino delegate from the app.

Tracked-On: OAM-128259
Signed-off-by: JeevakaPrabu <jeevaka.badrappan@intel.com>
Signed-off-by: Anoob Anto K <anoob.anto.kodankandath@intel.com>

sysopenci commented:
Improper Commit Message
Tracked on not found in commit message,
    make sure Tracked-On: Jira-ticket  is present.

sysopenci commented:
Improper Commit Message
Tracked on not found in commit message,
    make sure Tracked-On: Jira-ticket  is present.

sysopenci commented:
Improper Commit Message
Tracked on not found in commit message,
    make sure Tracked-On: Jira-ticket  is present.

sysopenci commented:
Improper Commit Message
Tracked on not found in commit message,
    make sure Tracked-On: Jira-ticket  is present.

ceiba1985 commented on npu/shell.te:
```diff
@@ -0,0 +1,3 @@
+allow shell npu_device:dir r_dir_perms;
```
you can wrap these two lines with userdebug_or_eng

akodanka committed:
Add userdebug_or_eng macro for shell sepolicy rules

Access to npu device via shel has to be restricted in user builds.

Changes done:
-Add userdebug_or_eng macro for shell sepolicy rules

Tests done:
-Verify hello_query_device does not work in user build.

Tracked-On: OAM-128259
Signed-off-by: Anoob Anto K <anoob.anto.kodankandath@intel.com>

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
SUCCESS: Android CI has completed Engineering Build for this issue.Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started MERGE Build for this pr ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed MERGE Build for this pr, build is SUCCESS. Please check the linked Tracked-On issue/Android CI Web  for more details. For Binaries: /cactus-absp-or-local/celadon-merge/882
